hybridse/src/plan/planner.cc

```diff
@@ -181,7 +203,25 @@ base::Status Planner::CreateSelectQueryPlan(const node::SelectQueryNode *root, P
     }
     // merge window map
     std::map<const node::WindowDefNode *, node::ProjectListNode *> merged_project_list_map;
-    CHECK_STATUS(MergeProjectMap(window_project_list_map, &merged_project_list_map))
+    bool long_window_exist = false;
+    if (!long_windows_.empty()) {
+        for (const auto &it : window_project_list_map) {
+            if (it.first == nullptr) continue;
+
+            if (long_windows_.count(it.first->GetName())) {
+                long_window_exist = true;
+                DLOG(INFO) << it.first->GetName() << " is long window. Disable project merge";
+                break;
+            }
+        }
+    }
+
+    if (long_window_exist) {
```

Comment:
What if users query with configuring a long window under BatchMode.

I suggest we only disable window merged for Request queried with long window option since the Physical Split Pass only happens to RequestMode transformation.

Reply:
Yes. You are right. Fixed.